tony-core/src/main/java/com/linkedin/tony/TonyClient.java

```diff
@@ -463,7 +464,8 @@ public void processTonyConfResources(Configuration tonyConf, FileSystem fs) thro
                       fs, LocalResourceType.FILE, jobName);
             }
           } else {
-            Utils.zipFolder(Paths.get(srcDir), Paths.get(file.getName()));
+            File tmpDir = Files.createTempDir();
+            Utils.zipFolder(Paths.get(resource), Paths.get(tmpDir.getAbsolutePath(), file.getName()));
```

Comment:
Now that you're creating the zip in `tmpDir`, I think you also need to update the `Utils. uploadFileAndSetConfResources()` call below to upload from the new location. Otherwise, it seems like the zip you created is not being used.

Can you also add a unit test for this?

Reply:
That sounds right..  @burgerkingeater

Reply:
How did it work when you tested it?